File: 0238-product-of-array-except-self/0238-product-of-array-except-self.js

var productExceptSelf = function(nums) {
    let left=new Array(nums.length);
    let right=new Array(nums.length);
    left[0]=1;
    right[nums.length-1]=1;
    for(let i=1;i<nums.length;i++){
        left[i]=left[i-1]*nums[i-1];
    }
    for(let i=nums.length-2;i>=0;i--){
        right[i]=right[i+1]*nums[i+1];
    }
    let ans=new Array(nums.length);
    for(let k=0;k<nums.length;k++){
        ans[k]=left[k]*right[k];
    }
    return ans;
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
// class Solution {
//     public int[] productExceptSelf(int[] nums) {
//         int pro=1;
//         for(int i=0;i<nums.length;i++){
//             pro=pro*nums[i];
//         }
//         int result[]=new int[nums.length];
//         for(int i=0;i<nums.length;i++){
//             if(nums[i]!=0){
//                 result[i]=pro/nums[i];
//             }else{
//                 result[i]=pro;
//             }
//         }
//         return result;
//     }
// }

class Solution {
    public int[] productExceptSelf(int[] nums) {
        int temp=1;
        int[] result=new int[nums.length];
        for(int i=0;i<nums.length;i++){
            result[i]=temp;
            temp=temp*nums[i];
        }
        temp=1;
        for(int i=nums.length-1;i>=0;i--){
            result[i]=result[i]*temp;
            temp=temp*nums[i];
        }
        return result;
    }
}